services/datcore-adapter/src/simcore_service_datcore_adapter/core/application.py

```diff
@@ -0,0 +1,56 @@
+import logging
+from typing import Optional
+
+from fastapi import FastAPI
+from models_library.basic_types import BootModeEnum
+
+from ..api.module_setup import setup_api
+from ..meta import api_version, api_vtag
+from ..modules import pennsieve
+from .events import (
+    create_start_app_handler,
+    create_stop_app_handler,
+    on_shutdown,
+    on_startup,
+)
+from .settings import Settings
+
+logger = logging.getLogger(__name__)
+
+
+def create_app(settings: Optional[Settings] = None) -> FastAPI:
+    if settings is None:
+        settings = Settings.create_from_envs()
+
+    logging.basicConfig(level=settings.DATCORE_ADAPTER_LOG_LEVEL.value)
+    logging.root.setLevel(settings.DATCORE_ADAPTER_LOG_LEVEL.value)
+
+    app = FastAPI(
+        debug=bool(
+            settings.SC_BOOT_MODE
+            in [BootModeEnum.DEBUG, BootModeEnum.DEVELOPMENT, BootModeEnum.LOCAL]
+        ),
+        title="Datcore Adapter Service",
+        description="Interfaces with Pennsieve storage service",
+        version=api_version,
+        openapi_url=f"/api/{api_vtag}/openapi.json",
+        docs_url="/dev/doc",
+        redoc_url=None,  # default disabled
+    )
+
+    logger.debug(settings)
+    app.state.settings = settings
+
+    # events
+    app.add_event_handler("startup", on_startup)
```

Comment:
well I guess that ensuring the correct order could be a nice add on. but I would also say that since we do not have too many of these dependencies that we should be able to live with that.